diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 7bbfe73..9cbe615 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -5,7 +5,7 @@ if(GETOPT_MISSING)
 endif()
 
 set(SIMPLE_EXAMPLES
-    osmium_convert
+    #osmium_convert
     osmium_count
     osmium_create_node_cache
     osmium_debug
diff --git a/include/osmium/util/double.hpp b/include/osmium/util/double.hpp
index 91f4ac7..629853b 100644
--- a/include/osmium/util/double.hpp
+++ b/include/osmium/util/double.hpp
@@ -37,6 +37,7 @@ DEALINGS IN THE SOFTWARE.
 #include <cassert>
 #include <cmath>
 #include <cstdio>
+#include <iterator>
 
 namespace osmium {
 
diff --git a/test/osm-testdata/CMakeLists.txt b/test/osm-testdata/CMakeLists.txt
index 743bb5a..2a0a695 100644
--- a/test/osm-testdata/CMakeLists.txt
+++ b/test/osm-testdata/CMakeLists.txt
@@ -29,6 +29,7 @@ endif()
 file(GLOB TESTCASE_CPPS testcases/*.cpp)
 add_executable(testdata-testcases testdata-testcases.cpp ${TESTCASE_CPPS})
 target_link_libraries(testdata-testcases
+                      ${OSMIUM_LIBRARIES}
                       ${OSMIUM_IO_LIBRARIES}
 )
 add_test(NAME testdata-testcases
@@ -41,6 +42,7 @@ set_tests_properties(testdata-testcases
 
 add_executable(testdata-overview testdata-overview.cpp)
 target_link_libraries(testdata-overview
+                      ${OSMIUM_LIBRARIES}
                       ${OSMIUM_IO_LIBRARIES}
                       ${GDAL_LIBRARIES}
 )
